### PR TITLE
Remove xray code for click to play

### PR DESCRIPTION
**Pull request by jonathanKingston**
**Reviewer:** @Charlie-belmer  or @kzar 

## Description:

This was fixed by https://bugzilla.mozilla.org/show_bug.cgi?id=1694679 so the regular code path works as intended.
https://app.asana.com/0/1176047645786601/1202155353194348/f

I've been testing on many sites and struggling to find the original breakage for this.

## Automated tests:
- [ ] Unit tests
- [ ] Integration tests

###### Reviewer Checklist:
- [ ] **Ensure the PR solves the problem**
- [ ] **Review every line of code**
- [ ] **Ensure the PR does no harm by testing the changes thoroughly**
- [ ] **Get help if you're uncomfortable with any of the above!**
- [ ] Determine if there are any quick wins that improve the implementation

###### PR Author Checklist:
- [ ] Get advice or leverage existing code
- [ ] Agree on technical approach with reviewer (if the changes are nuanced)
- [ ] Ensure that there is a testing strategy (and documented non-automated tests)
- [ ] Ensure there is a documented monitoring strategy (if necessary)
- [ ] Consider systems implications

### CODE REVIEW

File: integration-test/background/click-to-load-facebook.js [hunk 1]
```diff
@@ -159,8 +159,7 @@ describe('Facebook SDK schema', () => {
         //       to match
         //       /integration-test/artifacts/api_schemas/facebook-sdk.json
         //       and make any corresponding changes required to the surrogate
-        //       scripts /shared/data/web_accessible_resources/facebook-sdk.js
-        //       and /shared/js/content-scripts/fb-surrogate-xray.js.
+        //       script /shared/data/web_accessible_resources/facebook-sdk.js
```
Comment (kzar) [inline integration-test/background/click-to-load-facebook.js:162]: Nit: Missing full-stop.

**Review by kzar — COMMENTED** [review]
Thanks, great patch! I'm always happy to remove code like this!

Only thing is, [that Firefox bug was fixed with Firefox 89](https://bugzilla.mozilla.org/show_bug.cgi?id=1694679), but we so far support Firefox >= 65.0. I think this change also needs to disable CTP entirely for Firefox < 89 (or drop support for Firefox < 89).

FWIW my vote would be to drop support for Firefox < 89, but that might take some time to get approved :/.

**Comment by kzar** [comment]
@jonathanKingston Oh, one other thing. It would be nice to also remove the references to `fb-surrogate-xray.js` from `integration-test/data/configs/click-to-load-facebook.json` and `unit-test/data/clickToLoadConfig.json`.

**Comment by jonathanKingston** [comment]
> Only thing is, [that Firefox bug was fixed with Firefox 89](https://bugzilla.mozilla.org/show_bug.cgi?id=1694679), but we so far support Firefox >= 65.0. I think this change also needs to disable CTP entirely for Firefox < 89 (or drop support for Firefox < 89).

Right but they will still work in some contexts. IIRC it's only when the tag is marked with "crossorigin" and even then I think we'd basically just be blocking rather than having a semi working experience.

> Oh, one other thing. It would be nice to also remove the references to fb-surrogate-xray.js from integration-test/data/configs/click-to-load-facebook.json and unit-test/data/clickToLoadConfig.json.

That was actually intended to ensure that the code keeps working with the downloaded format. I think we should clear this up once we land this and then remove it from the remote file. Happy to remove before though.

**Comment by kzar** [comment]
> > Only thing is, [that Firefox bug was fixed with Firefox 89](https://bugzilla.mozilla.org/show_bug.cgi?id=1694679)...
> 
> Right but they will still work in some contexts. IIRC it's only when the tag is marked with "crossorigin" and even then I think we'd basically just be blocking rather than having a semi working experience.

Yea, I see what you mean. Fair enough, sounds good to me, sound OK to you @Charlie-belmer?
 
> > It would be nice to also remove the references to fb-surrogate-xray.js...
> 
> That was actually intended to ensure that the code keeps working with the downloaded format. I think we should clear this up once we land this and then remove it from the remote file. Happy to remove before though.

Ah right, got it. Hmm, I think my preference is still to remove those now.

**Review by kzar — APPROVED** [review]
LGTM :1st_place_medal:

**Comment by jonathanKingston** [comment]
> Ah right, got it. Hmm, I think my preference is still to remove those now.
Raised https://app.asana.com/0/1201720254973470/1202164530144430 to track the removal from the config

**Review by Charlie-belmer — APPROVED** [review]
LGTM

**Pull request merged**